tests/unit/test_scrape_jobs.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+import unittest
+from unittest.mock import PropertyMock, patch
+
+from alertmanager import WorkloadManager
+from charm import AlertmanagerCharm
+from helpers import k8s_resource_multipatch
+from ops.testing import Harness
+
+
+class TestWithInitialHooks(unittest.TestCase):
+    container_name: str = "alertmanager"
+
+    @patch("lightkube.core.client.GenericSyncClient")
+    @patch.object(WorkloadManager, "check_config", lambda *a, **kw: ("ok", ""))
+    @k8s_resource_multipatch
+    @patch.object(WorkloadManager, "_alertmanager_version", property(lambda *_: "0.0.0"))
+    def setUp(self, *unused):
+        self.harness = Harness(AlertmanagerCharm)
+        self.addCleanup(self.harness.cleanup)
+
+        self.harness.set_leader(True)
+        self.app_name = "alertmanager-k8s"
+        # Create the peer relation before running harness.begin_with_initial_hooks(), because
+        # otherwise it will create it for you and we don't know the rel_id
+        self.peer_rel_id = self.harness.add_relation("replicas", self.app_name)
+
+        self.harness.begin_with_initial_hooks()
+
+    @patch.object(AlertmanagerCharm, "_internal_url", new_callable=PropertyMock)
+    @patch.object(AlertmanagerCharm, "_scheme", new_callable=PropertyMock)
```

Comment:
Can we have the values specified in the decorator, i.e. without using PropertyMock at all? [Example](https://github.com/canonical/alertmanager-k8s-operator/blob/eda1588bf24d6343b9ad32e96ad9506ed3028c90/tests/unit/test_charm.py#L22).

This change would also make the `jobs_expected` more explicit, which I think is a plus.
I.e. repetitions like this in tests are ok.

Reply:
afaik the linked method doesn't work for mocking properties, only methods.  I'll give it a shot though